### ISSUE TITLE
NTLK Download Error

### ISSUE THREAD

**Issue by otaku0304** [nltk/nltk#3092]
Using nltk download function an error was thrown ( Connecting to server error ).  

![error](https://user-images.githubusercontent.com/107242846/209286811-bbefeea8-a4b0-4b35-9830-6281522d7216.png)

**Comment by adityak74** [comment]
I am seeing the same error on mac as well.

**Comment by r9922** [comment]
SAME for me i tried from last 2 days now lots time i have spend but no soution found on internet,even when search link its also invalid
![10060](https://user-images.githubusercontent.com/121342205/209434318-7e1fa2b1-9400-491c-a5e1-e80397b96adb.PNG)
![link](https://user-images.githubusercontent.com/121342205/209434322-9d17e7a5-4203-48cf-b47c-31e8d2397329.PNG)

**Comment by r9922** [comment]
otaku0304 & adityak74 
HI,
if you are using wifi of any network then try on mobile hotspot network. i tried on BSNL network on mobile hotspot now its working and also downloading packages .ONLY FOR INDIA network issue
![image](https://user-images.githubusercontent.com/121342205/209437015-5f2586c6-a7c0-4885-a15b-253fbe76f4d9.png)
![image](https://user-images.githubusercontent.com/121342205/209437034-64f8d31f-7cef-4798-b23c-de9e9780430e.png)

**Comment by r9922** [comment]
if working let me know in comment.

**Comment by r9922** [comment]
also you can try the VPN.

**Comment by otaku0304** [comment]
Sure brother.

**Comment by fievelk** [comment]
I can't seem to reproduce it. I did a clean install of NLTK and running `nltk.download()` seems to be working fine. Do you have other steps or information that could help reproduce the issue?

**Comment by tomaarsen** [comment]
This seems to be related to raw.githubusercontent.com sometimes being down for some users. I think all we can do here is to wait for GitHub to resolve these issues. I believe the outages are intermittent, so waiting a bit before retrying may help. 

Alternatively, you can use https://www.nltk.org/nltk_data/index.xml as the NLTK Data Server Index, either at the bottom of the `nltk.download()` UI, or by specifying the `-u` or `--url` parameter in the NLTK CLI downloader:
* `python -m nltk.downloader wordnet -u https://www.nltk.org/nltk_data/index.xml` to download wordnet, as shown on the downloader.py options (`python -m nltk.downloader --help`):
   ```
   Usage: downloader.py [options]
   
   Options:
     -h, --help            show this help message and exit
     -d DIR, --dir=DIR     download package to directory DIR
     -q, --quiet           work quietly
     -f, --force           download even if already installed
     -e, --exit-on-error   exit if an error occurs
     -u SERVER_INDEX_URL, --url=SERVER_INDEX_URL
                           download server index url
   ```

Hope that helps!

**Comment by otaku0304** [comment]
> otaku0304 & adityak74 HI, if you are using wifi of any network then try on mobile hotspot network. i tried on BSNL network on mobile hotspot now its working and also downloading packages .ONLY FOR INDIA network issue ![image](https://user-images.githubusercontent.com/121342205/209437015-5f2586c6-a7c0-4885-a15b-253fbe76f4d9.png) ![image](https://user-images.githubusercontent.com/121342205/209437034-64f8d31f-7cef-4798-b23c-de9e9780430e.png)

@r9922 Brother it is working while using VPN. But the problem is the same when using the Indian Network.

**Comment by otaku0304** [comment]
> This seems to be related to raw.githubusercontent.com sometimes being down for some users. I think all we can do here is to wait for GitHub to resolve these issues. I believe the outages are intermittent, so waiting a bit before retrying may help.
> 
> Alternatively, you can use https://www.nltk.org/nltk_data/index.xml as the NLTK Data Server Index, either at the bottom of the `nltk.download()` UI, or by specifying the `-u` or `--url` parameter in the NLTK CLI downloader:
> 
> * `python -m nltk.downloader wordnet -u https://www.nltk.org/nltk_data/index.xml` to download wordnet, as shown on the downloader.py options (`python -m nltk.downloader --help`):
>   ```
>   Usage: downloader.py [options]
>   
>   Options:
>     -h, --help            show this help message and exit
>     -d DIR, --dir=DIR     download package to directory DIR
>     -q, --quiet           work quietly
>     -f, --force           download even if already installed
>     -e, --exit-on-error   exit if an error occurs
>     -u SERVER_INDEX_URL, --url=SERVER_INDEX_URL
>                           download server index url
>   ```
> 
> Hope that helps!

@tomaarsen  What you said is correct we need to wait for the team to fix the issue. People who want to download this can by using VPN.

**Comment by tomaarsen** [comment]
Wonderful. Glad you got it working with that workaround.

**Comment by fievelk** [comment]
Just out of curiosity: what's your network provider? I'm reading reports of people having issues reaching raw.githubusercontent.com using Jio. They seem to be DNS issues.

**Comment by lokesh074** [comment]
i am also facing same issue i connected with mobile hotspot but not able to install and manually download site also not working

**Comment by tomaarsen** [comment]
@lokesh074 Perhaps the workaround from https://github.com/nltk/nltk/issues/3092#issuecomment-1366526151 works for you? Or have you also tried that already?

When going to https://raw.githubusercontent.com/nltk/nltk_data/gh-pages/index.xml I don't have any issues myself, but it does appear to be down according to websites that track whether websites are down:
![image](https://user-images.githubusercontent.com/37621491/209946335-41193db9-7bd6-4671-9b97-a5348b1b7780.png)

**Comment by 1934001** [comment]
It seems to look like jio DNS error., I tried to download the package using Airtel network and it works well and good

**Comment by Rohit-33** [comment]
Yes guys please use different dns servicer like BSNL , AIRTEL it will be fine.

**Comment by chintanckg** [comment]
I faced the same issue today. It's not about wifi/hotspot. You need to use VPN.

Hoping the issue will be fixed soon, meanwhile, VPN is the simplest solution.

**Comment by AnkitDubey040** [comment]
Wow it really helped , Thanks

**Comment by bevsxyz** [comment]
> This seems to be related to raw.githubusercontent.com sometimes being down for some users. I think all we can do here is to wait for GitHub to resolve these issues. I believe the outages are intermittent, so waiting a bit before retrying may help.
> 
> Alternatively, you can use https://www.nltk.org/nltk_data/index.xml as the NLTK Data Server Index, either at the bottom of the `nltk.download()` UI, or by specifying the `-u` or `--url` parameter in the NLTK CLI downloader:
> 
>     * `python -m nltk.downloader wordnet -u https://www.nltk.org/nltk_data/index.xml` to download wordnet, as shown on the downloader.py options (`python -m nltk.downloader --help`):
>       ```
>       Usage: downloader.py [options]
>       
>       Options:
>         -h, --help            show this help message and exit
>         -d DIR, --dir=DIR     download package to directory DIR
>         -q, --quiet           work quietly
>         -f, --force           download even if already installed
>         -e, --exit-on-error   exit if an error occurs
>         -u SERVER_INDEX_URL, --url=SERVER_INDEX_URL
>                               download server index url
>       ```
> 
> 
> Hope that helps!

I think this is closer to the solution that worked for me.
https://github.com/nltk/nltk/issues/1787#issuecomment-318256965

```bash
wget https://github.com/nltk/nltk_data/archive/gh-pages.zip
unzip gh-pages.zip
mkdir ~/nltk_data
mv nltk_data-gh-pages/nltk_data-gh-pages/packages/* nltk_data/
cd nltk_data/tokenizers/
unzip punkt.zip
```

**Comment by HywelMJ** [comment]
python
import nltk
nltk.download()

gives me a similar error: WinError10054

![Screenshot 2023-03-27 180645](https://user-images.githubusercontent.com/3739082/228014932-5bf15168-68da-4452-9bcb-677d23e898d4.png)

Changing the NLTK Data Server Index, to https://www.nltk.org/nltk_data/index.xml makes no difference, nor does changing the download directory.

**Comment by bevsxyz** [comment]
> > This seems to be related to raw.githubusercontent.com sometimes being down for some users. I think all we can do here is to wait for GitHub to resolve these issues. I believe the outages are intermittent, so waiting a bit before retrying may help.
> > 
> > Alternatively, you can use https://www.nltk.org/nltk_data/index.xml as the NLTK Data Server Index, either at the bottom of the `nltk.download()` UI, or by specifying the `-u` or `--url` parameter in the NLTK CLI downloader:
> > 
> >     * `python -m nltk.downloader wordnet -u https://www.nltk.org/nltk_data/index.xml` to download wordnet, as shown on the downloader.py options (`python -m nltk.downloader --help`):
> >       ```
> >       Usage: downloader.py [options]
> >       
> >       Options:
> >         -h, --help            show this help message and exit
> >         -d DIR, --dir=DIR     download package to directory DIR
> >         -q, --quiet           work quietly
> >         -f, --force           download even if already installed
> >         -e, --exit-on-error   exit if an error occurs
> >         -u SERVER_INDEX_URL, --url=SERVER_INDEX_URL
> >                               download server index url
> >       ```
> > 
> > 
> > Hope that helps!
> 
> I think this is closer to the solution that worked for me.
> https://github.com/nltk/nltk/issues/1787#issuecomment-318256965
> 
> ```bash
> wget https://github.com/nltk/nltk_data/archive/gh-pages.zip
> unzip gh-pages.zip
> mkdir ~/nltk_data
> mv nltk_data-gh-pages/nltk_data-gh-pages/packages/* nltk_data/
> cd nltk_data/tokenizers/
> unzip punkt.zip
> ```

@HywelMJ you may get it working with manually downloding the archive from GitHub and populating files under the `nltk_data` folder similar to the linux commands I've shared.
